src/prefect/server/api/workers.py

```diff
@@ -312,14 +315,15 @@ async def get_scheduled_flow_runs(
             session=session, work_pool_name=work_pool_name
         )
 
-        if work_queue_names is None:
+        if not work_queue_names:
             work_queue_ids = None
-            ready_work_queue_ids = [
+            polled_work_queue_ids = [
                 wq.id
                 for wq in await models.workers.read_work_queues(
                     session=session, work_pool_id=work_pool_id
                 )
             ]
+            ready_work_queue_ids = polled_work_queue_ids
```

Comment:
Could this be the culprit that's causing all polled work queues to be emitting ready events? Seems like we'd need to filter these

Reply:
It was!  https://github.com/PrefectHQ/prefect/pull/13117